libs/klv/src/main/java/org/codice/alliance/libs/klv/GeoBoxHandler.java

```diff
@@ -156,6 +151,18 @@ public String getLongitude4() {
         return asAttribute(polygonsWkts);
     }
 
+    /**
+     * Trim the arrays of lat and lon values to the same length.
+     */
+    @Override
+    public void trim() {
+        Trimmable.trimList(map.values(), getMinimumListSize());
```

Comment:
This functionality concerns me a great deal. I know I have a greater dislike for mutable objects/collections than many other engineers, but you're fiddling with the contents of a collection on the fly in a really gnarly way. Clearly this class is not threadsafe and, I assume, is not intended to be used in any way where it could be shared by threads; however, this is still awfully...bad.

If you were making deep copies and returning immutable collections that would be great; performing surgery on the contents of those collections is just worrisome. I'm confident we can find a better way.